packages/react-native/scripts/ios-prebuild.js

```diff
@@ -0,0 +1,140 @@
+   * @throws {Error} If the source path does not exist or if the link creation fails
+   * @returns {void}
+   */
+  const link = (
+    fromPath /*:string*/,
+    toPath /*:string*/,
+    includePath /*:string*/,
+  ) => {
+    if (toPath.includes('__tests__')) {
+      // Skip test folders
+      return;
+    }
+
+    console.log(`Linking ${fromPath} to ${toPath + '/' + includePath}...`);
+    const source = path.resolve(root, fromPath);
+    const target = path.resolve(linksFolder, toPath, includePath);
+
+    // get subfolders in source
+    const entries = fs.readdirSync(source, {withFileTypes: true});
+    if (
+      entries.some(
+        dirent =>
+          dirent.isFile() &&
+          (dirent.name.endsWith('.h') || dirent.name.endsWith('.hpp')),
+      )
+    ) {
+      // Create link for current folder
+      createLink(source, target);
+    }
+
+    const subfolders = entries
+      .filter(dirent => dirent.isDirectory())
+      .map(dirent => dirent.name);
+
+    // Create links for subfolders
+    subfolders.forEach(folder => {
+      link(fromPath + '/' + folder, toPath + '_' + folder, includePath);
+    });
+  };
+
+  link('Libraries/WebSocket/', 'WebSocket', 'React');
+  link('React/Base', 'Base', 'React');
+  link('React/Base/Surface', 'Surface', 'React');
+  link('React/CxxBridge', 'CxxBridge', 'React');
+  link('React/CxxModule', 'CxxModule', 'React');
+  link('React/CxxUtils', 'CxxUtils', 'React');
+  link('React/DevSupport', 'DevSupport', 'React');
+  link('React/Inspector', 'Inspector', 'React');
+  link('React/I18n', 'I18n', 'React');
+  link('React/Views', 'Views', 'React');
+  link('React/CoreModules', 'CoreModules', 'React');
+  link('React/Modules', 'Modules', 'React');
+  link('React/Fabric', 'Fabric', 'React');
+  link('React/Profiler', 'Profiler', 'React');
+  link('React/CoreModules', 'CoreModules', 'React');
+
+  link('React/Runtime', 'Runtime', 'React');
+  link('React/Views/ScrollView', 'ScrollView', 'React');
+  link('React/Views/RefreshControl', 'RefreshControl', 'React');
+
+  link(
+    'ReactApple/Libraries/RCTFoundation/RCTDeprecation/Exported',
+    'ReactApple/',
+    'RCTDeprecation',
+  );
+  link('Libraries/Required', 'Required', 'RCTRequired');
+  link('Libraries/TypeSafety', 'TypeSafety', 'RCTTypeSafety');
+  link('Libraries/Text', 'Text', 'React');
+  link('Libraries/Image', 'Image', 'React');
+  link('Libraries/Network', 'Network', 'React');
+  link('Libraries/Blob', 'Blob', 'React');
+  link('Libraries/NativeAnimation', 'NativeAnimation', 'React');
+
+  link('ReactCommon/hermes', 'Hermes', 'reacthermes');
+  link('ReactCommon/hermes', 'Hermes', 'jsireact');
+
+  link(
+    'ReactCommon/react/renderer/imagemanager',
+    'ImageManager',
+    'react/renderer/imagemanager',
+  );
+
+  console.log('Running codegen...');
+  execSync(
+    `node scripts/generate-codegen-artifacts -p "${process.cwd()}" -t ios -o "${process.cwd()}/.build/codegen" -s library`,
+  );
```

Comment:
I think we should run codegen before creating the links. Codegen adds some files that in the future might need to be linked.